modules/docs/arrow-docs/docs/docs/typeclasses/profunctor/README.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: docs
+title: Profunctor
+permalink: /docs/typeclasses/profunctor/
+---
+
+## Profunctor
+
+{:.intermediate}
+intermediate
+
+Before reading this typeclass we recommend you to understand [Contravariance](https://typeclasses.com/contravariance) first. But for making things easier we will consider `Contravariance` as the ability to _flip composition_.
+
+`Profunctors` are [`Bifunctors`]({{ '/docs/typeclasses/bifunctor' | relative_url }}) that are contravariant in their first argument and covariant in the second one.
+
+The core operation of the `Profunctor` typeclass is `dimap` (as `bimap` was already taken for `Bifunctor`).
+
+`fun <A, B, C, D> Kind2<F, A, B>.dimap(fl: (C) -> A, fr: (B) -> D): Kind2<F, C, D>`
+
+The main difference between `bimap` and `dimap` is the function they accept as their first argument:
+
+* `bimap`: `fl: (A) -> C`
+* `dimap`: `fl: (C) -> A`
+
+And how this works? Well, if we think in terms of function composition, functions can be composed in both directions:
+
+```kotlin:ank
+val sum2: (Int) -> Int = { x -> x + 2 }
+val str: (Int) -> Int = { x -> x.toString() }
+
+val f = str compose sum2
+val g = sum2 andThen str
+f(4) == g(4) 
+```
+
+Functions are a binary type constructor of an input type and an output type. It is implemented in [`Function1`]({{ '/docs/datatypes/function1' | relative_url }}).
+
+So, if we have a function `(A) -> B` and a `Profunctor` instance for it, we can make the following transformation with `bimap`: `((C) -> A) -> ((A) -> B) -> ((B) -> D)`.
+
+Example:
+
+```kotlin:ank
+import arrow.core.*
+import arrow.instances.*
+
+val fab: Function1<Double, String> = { x -> "Result: $x" }.k()
+val f: (Int) -> Double = { x -> x.toDouble() / 2 }  
+val g: (Double) -> Double = { x -> x * 3 } 
+
+val h: Function1<Int, String> = Function1.profunctor().run { fab.bimap(f, g) } 
+h(4)
+```
+
+### Main Combinators
+
+#### Kind2<F, A, B>#bimap
+
+Contramap on the first type parameter and map on the second type parameter
+
+`fun Kind2<F, A, B>.dimap(fl: (C) -> A, fr: (B) -> D): Kind2<F, C, D>`
+
```

Comment:
Add one snippet here, please

Reply:
Okay, but it can be similar to the example I provided so it can be repetitive.